packages/core/src/baseclient.ts

```diff
@@ -582,7 +654,7 @@ export abstract class BaseClient<B extends Backend, O extends Options> implement
           this._updateSessionFromEvent(session, processedEvent);
         }
 
-        this._sendEvent(processedEvent);
+        this.sendEvent(processedEvent);
```

Comment:
```suggestion
        this._sendEvent(processedEvent);
```

This should have stayed `_sendEvent`, which then itself will call `sendEvent`. By bypassing `_sendEvent`, you're also bypassing the `super._sendEvent` call, so event breadcrumbs are never getting attached. That's why your tests are failing.

Reply:
Good catch! Let's make sure we address this when we get to reviewing the client inheritance chain + looking at all the indirection in the SDK (part 6). We can maybe make the event processing chain smaller and easier to understand.

I made some tickets that went over this.